include/seqan3/alignment/pairwise/detail/policy_optimum_tracker.hpp

```diff
@@ -0,0 +1,173 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::policy_optimum_tracker.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
```

Comment:
well, I guess this is one of the controversial features discussed in the community. 
The first thing is, that it is not a standard feature but so far a well supported one by the major compiler vendors. At least the one we promise to work with.
Another issues that arised in the past is that the pragma did had issues with large code bases where it can happen that the same header base name (`a/foo.hpp` and `b/foo.hpp`) was used in different locations. However, that is also a problem for include guards (but of course can be fixed manuualy) but it was fixed already quite some time ago. I am not aware of other issues that might happen though.
This means as far as I know there is no disadvantage of `#pragma once` over the include guards. At least we didn't encounter one yet. Not sure if it comiles faster, which is indicated by some threads. However, the general advantage over include guards is spelling safety:
```
#indef some_intermediate_header_with_a_long_name
#define some_intermidiate_header_with_a_long_name
...
#endif // some_intermediate_header_with_a_long_name
```
So as you can see, not only is it much easier to type but also prevents some very hard to track errors which could literally blow the compile time and the binary size if the header is included many many times or end in some weird not so obvious to fix error messages because of redefinitions etc.. Similarly to have a non matching pair of parenthesis in the code which gives you suddenly errors about some namespaces not defined in some completely random files ;)
So our policy is to go with `#pragma once` as long as it does not raise any troubles, which seems to be very unlikely and hope to live happily ever after. 😄

Reply:
Thank you for the elaborate explanation!